src/containerapp/azext_containerapp/_ssh_utils.py

```diff
@@ -0,0 +1,57 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=logging-fstring-interpolation
+# pylint: disable=possibly-used-before-assignment
+
+from azure.cli.command_modules.containerapp._ssh_utils import WebSocketConnection, SSH_TERM_RESIZE_PREFIX, \
+    SSH_DEFAULT_ENCODING, read_ssh
+from azure.cli.core.azclierror import ValidationError, CLIInternalError
+
+from knack.log import get_logger
+
+from azext_containerapp._clients import ContainerAppPreviewClient
+
+logger = get_logger(__name__)
+
+
+class DebugWebSocketConnection(WebSocketConnection):
+    def __init__(self, cmd, resource_group_name, name, revision, replica, container):
+        try:
+            super(DebugWebSocketConnection, self).__init__(cmd, resource_group_name,
+                                                           name, revision, replica, container, "")
+        except Exception as e:
+            stringErr = str(e)
+            if "Handshake status 400 Bad Request" in stringErr:
+                raise CLIInternalError("Debug Console feature is not allowed for the subscription.")
```

Comment:
400 doesnt necessarily mean a feature flag issue. might be better to show detailed error messages or have an error code mapping

Reply:
the python websocket library didn't read response body if response code is not in (301, 302, 303 and 101). So we can't read the body and only can read 400 reponse code.

Given we will remove feature flag in the future, let me remove this logic.

Reply:
BTW, the newest websocket client support get response body: https://github.com/websocket-client/websocket-client/blob/master/websocket/_handshake.py#L146